File: OctobridgeEvaluation/Angular.Client/src/app/models/orderdetail.ts
//
// OrderDetail Class
//
export class OrderDetail {
  lineId: number;
  orderId: number;
  productId: number;
  unitPrice: number;
  quantity: number;
  recordCreated: Date | null;
  recordModified: Date | null;

  constructor(
    theLineId = -1,
    theOrderId = -1,
    theProductId = -1,
    theUnitPrice = -1,
    theQuantity = -1,
    theRecordCreated: Date | null = null,
    theRecordModified: Date | null = null
  ) {
    this.lineId = theLineId;
    this.orderId = theOrderId;
    this.productId = theProductId;
    this.unitPrice = theUnitPrice;
    this.quantity = theQuantity;
    this.recordCreated = theRecordCreated;
    this.recordModified = theRecordModified;
  }
}

File: OctobridgeEvaluation/Angular.Server/Models/OrderDetail.cs
using System;

namespace AngularClient.Models
{
    public class OrderDetail
    {
        public int LineId { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public decimal UnitPrice { get; set; }
        public int Quantity { get; set; }
        public DateTime? RecordCreated { get; set; }
        public DateTime? RecordModified { get; set; }

        public OrderDetail()
        {
        }
    }
}
